File: Triangle/Triangle.py
class Solution(object):
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        res = triangle[-1]
        for i in range(len(triangle) - 2, -1, -1):
            for j in range(i + 1):
                res[j] = min(res[j], res[j + 1]) + triangle[i][j]
        return res[0]

File: Triangle/Triangle.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        if(triangle.size() == 0)
        	return 0;
        vector<int> dp = triangle.back();
        int size = triangle.size();
        for(int i = size - 2; i >= 0; i--) {
        	for(int j = 0; j < triangle[i].size(); j++) {
        		dp[j] = min(dp[j], dp[j + 1]) + triangle[i][j];
        	}
        }
        return dp[0];
    }
};